src/kinect_fusion_opencl_pose_estimation_pipeline_block.cpp

```
@@ -71,15 +71,21 @@ namespace dynfu {
 
 		}
 
-		auto frame_size_after=frame_size/group_size_;
-		mats_=boost::compute::buffer(q_.get_context(),frame_size_after*sizeof_mats);
-		if ((frame_size_after%group_size_)==0) mats_output_=boost::compute::buffer(q_.get_context(),(frame_size_after/group_size_)*sizeof_mats);
-
 		auto p=pf("pose_estimation");
 		corr_=p.create_kernel("correspondences");
 		parallel_sum_=p.create_kernel("parallel_sum");
 		serial_sum_=p.create_kernel("serial_sum");
 
+		group_size_ = std::min(group_size_, corr_.get_work_group_info<std::size_t>(q_.get_device(), CL_KERNEL_WORK_GROUP_SIZE));
```

Don't do this.  This should be handled outside this class.

Not just for the sake of the unit tests for degenerate group sizes, but also because if the user makes a request (for a work group size) we should honour it and throw if it's not possible rather than silently and mysteriously not honouring their request.